anta/tests/hardware.py

```diff
@@ -694,20 +700,28 @@ def _verify_specific_requirements(self, inventory: dict[str, dict[str, Any]]) ->
             component_data = inventory[component_type]
 
             if requirement == "all":
+                # atomic results
+                result = self.result.add(description=f"All available slots for {component_type.replace('_', ' ').title()}")
+                result.is_success()
                 # All available slots for this component type must be inserted
-                self._report_failures(component_data)
+                self._report_failures(component_data, result)
                 continue
 
+            # atomic results for specific requirements
+            result = self.result.add(description=f"{component_type.replace('_', ' ').title()}")
+            result.is_success()
             if isinstance(requirement, int) and (installed_count := component_data["installed"]) < requirement:
                 # Check if the number of installed units meets the minimum requirement
-                self.result.is_failure(f"{component_type.replace('_', ' ').title()} - Count mismatch - Expected: >= {requirement} Actual: {installed_count}")
+                result.is_failure(f"Count mismatch - Expected: >= {requirement} Actual: {installed_count}")
 
-    def _report_failures(self, component_data: dict[str, Any]) -> None:
+    def _report_failures(self, component_data: dict[str, Any], result: AtomicTestResult) -> None:
         """Report failures for a given component type based on its state."""
         for slot in component_data.get("not_inserted", []):
-            self.result.is_failure(f"{slot} - Not inserted")
+            result.description = slot
+            result.is_failure("Not inserted")
```

Comment:
Here we are adding multiple failure messages to the same atomic result which must keep the same description (component type we are checking). We simply need to add each failure message with added context.
```suggestion
            result.is_failure(f"{slot} - Not inserted")
```